### ISSUE TITLE
run: make --name a regular argument?

### ISSUE THREAD

**Issue by jorgeorpinel** [iterative/dvc#4172]
The current minimal use of `dvc run` requires both `--name/-n` (option) and a `command` (argument). I believe `-n` is in fact one of the few or maybe the only required option in all of our commands. How about make it into a proper argument?

```console
$ dvc run [--options] <name> <command>
```

E.g.

```console
$ dvc run train -d data -d train.py -o model python train.py
```

We'd may need to keep `-n` around (as an option to the `name` arg) for backward compatibility, though.

---

Another idea is to allow `-d/--deps` and `-o/--outs` to accept several option args, to avoid repeating the flags. This already applies to `-p/--params`, via commas.

E.g.

```console
$ dvc run featurize \
       -d data/clean,feats.py \
       -p f.range,f.limit \
       -o data/feats,data/map.csv \
          /feats.py
```

**Comment by efiop** [comment]
@jorgeorpinel Great idea! We didn't do that yet because `dvc run` is too overloaded already and I hope that we will have something like `dvc stage add`(or something else) in the future, where it would have a more friendly interface. Let's keep this issue open for now.

**Comment by jorgeorpinel** [comment]
Agree. `dvc stage [--options] <name> <cmd>` to create/update (maybe even `--remove` instead of `dvc remove`) stages, and separately `dvc run` to run/repro stages and pipelines seems ideal for the 1.0 era IMO.

**Comment by efiop** [comment]
Closing as stale
